src/Microsoft.Health.Encryption/Customer/Health/CustomerKeyValidationBackgroundService.cs

```diff
@@ -0,0 +1,105 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using EnsureThat;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Core.Features.Health;
+using Microsoft.Health.Encryption.Customer.Configs;
+
+namespace Microsoft.Health.Encryption.Customer.Health;
+
+internal class CustomerKeyValidationBackgroundService : BackgroundService
+{
+    private const string AccessLostMessage = "Access to the customer-managed key has been lost";
+    private const int DefaultKeyValidationPeriodInSeconds = 60;
+
+    private readonly IKeyTestProvider _keyTestProvider;
+    private readonly ValueCache<CustomerKeyHealth> _customerManagedKeyHealth;
+    private readonly ILogger<CustomerKeyValidationBackgroundService> _logger;
+
+    private readonly TimeSpan _keyValidationPeriod;
+
+    public CustomerKeyValidationBackgroundService(
+        IKeyTestProvider keyTestProvider,
+        ValueCache<CustomerKeyHealth> customerManagedKeyHealth,
+        IOptions<CustomerManagedKeyOptions> customerManagedKeyOptions,
+        ILogger<CustomerKeyValidationBackgroundService> logger)
+    {
+        EnsureArg.IsNotNull(customerManagedKeyOptions, nameof(customerManagedKeyOptions));
```

Comment:
Assert that `customerManagedKeyOptions?.Value` is not `null`

Reply:
Ah I had assumed that if a consumer didn't specify these cmk options that `customerManagedKeyOptions?.Value` would be null, and we had to handle that case. But looks like it just creates an empty object